client/src/components/profile/components/TimeLine.js

```diff
@@ -81,9 +108,39 @@ class TimelineInner extends Component {
     }));
   }
 
+  firstPage() {
+    this.setState({
+      pageNo: 1
+    });
+  }
+  nextPage() {
+    this.setState(state => ({
+      pageNo: state.pageNo + 1
+    }));
+  }
+
+  prevPage() {
+    this.setState(state => ({
+      pageNo: state.pageNo - 1
+    }));
+  }
+  lastPage() {
+    this.setState({
```

Comment:
```suggestion
    this.setState((_, props) => ({
      pageNo: props.totalPages
    }));
```
Just to be extra careful, since `setState` calls can be batched.